src/internal_modules/roc_core/memory_limiter.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/memory_limiter.h
+//! @brief Memory limiter.
+
+#ifndef ROC_CORE_MEMORY_LIMITER_H_
+#define ROC_CORE_MEMORY_LIMITER_H_
+
+#include "roc_core/atomic.h"
+#include "roc_core/noncopyable.h"
+#include "roc_core/stddefs.h"
+
+namespace roc {
+namespace core {
+
+//! Memory limiter.
+class MemoryLimiter : public NonCopyable<> {
+public:
+    //! Initialize memory limiter.
+    explicit MemoryLimiter(const char* name, size_t max_bytes);
```

Comment:
Could you please add a small comment, describing:

* the purpose of the class (why we need it / how we use it)
* meaning of acquire/release and max_bytes (including zero value)
* contracts (can't release more than acquired, should release all before dtor)